controllers/topology/blueprint.go

```diff
@@ -65,6 +65,11 @@ func (r *ClusterReconciler) getBlueprint(ctx context.Context, cluster *clusterv1
 		}
 	}
 
+	// If the clusterClass defines a valid MachineHealthCheck (including a defined MachineInfrastructure) set the blueprint MachineHealthCheck.
+	if blueprint.HasControlPlaneMachineHealthCheck() {
+		blueprint.ControlPlane.MachineHealthCheck = blueprint.ClusterClass.Spec.ControlPlane.MachineHealthCheck
```

Comment:
Exactly - this was done to make it so the MachineHealthChecks could be accessed through the MDClass rather than needing to do some matching for the name. For the ControlPlane it's not importan

Reply:
Sounds good!